dnsimple/webhook/events.go

```diff
@@ -173,6 +173,32 @@ func (d *DomainEventData) unmarshalEventData(payload []byte) error {
 	return unmarshalEventData(payload, d)
 }
 
+//
+// DomainLockEvent
+//
+
+// DomainLockEventData represents the data node for a DomainLock event.
+type DomainLockEventData struct {
+	Domain     *dnsimple.Domain     `json:"domain"`
+}
+
+func (d *DomainLockEventData) unmarshalEventData(payload []byte) error {
+	return unmarshalEventData(payload, d)
+}
+
+//
+// DomainUnlockEvent
+//
+
+// DomainUnlockEventData represents the data node for a DomainUnlock event.
+type DomainUnlockEventData struct {
+	Domain     *dnsimple.Domain     `json:"domain"`
+}
+
+func (d *DomainUnlockEventData) unmarshalEventData(payload []byte) error {
+	return unmarshalEventData(payload, d)
+}
+
```

Comment:
Since the `domain.lock` and `domain.unlock` are domain events i.e. they have a common schema without any custom data, they can be directly added to the domain section without implementing their own types.

Reply:
So the additional fields in `DomainEventData` are fine?

Reply:
Sorry, you are right the schema does not match. Resolving.